syntax = "proto3";

service Wallet {
    rpc Balance(BalanceRequest) returns (BalanceResponse) {}
    rpc SalesOrder(SalesOrderRequest) returns (SalesOrderResponse) {}
    rpc Transfer(TransferRequest) returns (TransferResponse) {}
    rpc End(EndRequest) returns (EndResponse) {}
}

message BalanceRequest {
    string account_id = 1;
}

message BalanceResponse {
    int32 balance = 2;
}

message SalesOrderRequest {
    string account_id = 1;
    int32 value = 2;
}

message SalesOrderResponse {
    int32 status = 1;
    int32 order_id = 2;
}

message TransferRequest {
    int32 order_id = 1;
    int32 value = 2;
    string account_id = 3;
}

message TransferResponse {
    int32 status = 1;
}

message EndRequest {}

message EndResponse {
    int32 pendencies = 1;
}
